contracts/DutchExchange.sol

```diff
@@ -463,34 +464,47 @@ contract DutchExchange {
 
         // We have to compute the price at intersection time,
         // which will be exactly half of initial price
-        fraction memory sellTokenPrice = priceOracle(sellToken);
-        fraction memory buyTokenPrice = priceOracle(buyToken);
-        fraction memory price;
-
-        price.num = sellTokenPrice.num * buyTokenPrice.den;
-        price.den = sellTokenPrice.den * buyTokenPrice.num;
+        fraction memory price = computeRatioOfHistoricalPriceOracles(sellToken, buyToken, auctionIndex);
```

Comment:
Can you please explain? the oustandingVolume is calculated with the getPrice of the current time. Then we can not calculate the outstandingVolumeOpp with a completely new price, right?

Reply:
Hm thought about it again. Probably we should calculate both oustandingVolume and outstandingVOlumeOpp with the priceOracle.

Reply:
That's exactly what this does ;) I just reduced code duplication by extracting it into its own function (since getPrice uses the same mechanism).

Reply:
I think you did this only for the outstandingVolumeOpp, but not for the outstandingVolume. This one is still calculated within the postBuyOrder with getPrice.

Reply:
It's quite confusing to so many different price function, which all call another pricefunction. Though I am not sure, whether we can reduce complexity

Reply:
I'm with you one the last one, but solving bugs with the smallest possible complexity is what I do 9 to 5 :D 
I'll think about the first comm, if it should be different